MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -409,89 +414,85 @@ namespace WPEFramework {
         }
 
 #if defined(ENABLE_WHOAMI)
-        bool MaintenanceManager::knowWhoAmI(string &activation_status)
+        bool MaintenanceManager::knowWhoAmI()
         {
             bool success = false;
-            int retryDelay = 10;
-            int retryCount = 0;
+            int paramEmpty = false;
             const char* secMgr_callsign = "org.rdk.SecManager";
             const char* secMgr_callsign_ver = "org.rdk.SecManager.1";
             PluginHost::IShell::state state;
             WPEFramework::JSONRPC::LinkType<WPEFramework::Core::JSON::IElement>* thunder_client = nullptr;
 
-            do {
+            
+            if ((getServiceState(m_service, secMgr_callsign, state) == Core::ERROR_NONE) && (state == PluginHost::IShell::state::ACTIVATED)) 
+            {
+                LOGINFO("%s is active", secMgr_callsign);
+                thunder_client=getThunderPluginHandle(secMgr_callsign_ver);
+                if (thunder_client != nullptr) 
+                {
+                    JsonObject params;
+                    JsonObject joGetResult;
+
+                    thunder_client->Invoke<JsonObject, JsonObject>(5000, "getDeviceInitializationContext", params, joGetResult);
+                    if (joGetResult.HasLabel("success") && joGetResult["success"].Boolean()) 
+                    {
+                        static const char* kDeviceInitializationContext = "deviceInitializationContext";
+                        if (joGetResult.HasLabel(kDeviceInitializationContext)) 
+                        {
+                            JsonObject getInitializationContext = joGetResult[kDeviceInitializationContext].Object();
```

Comment:
A separate method to take in secManager response JsonObject, parse them Set RFC and then AuthService (setPartnerId). Which will be called only after the event is received (here it is after thread notify), in case of WhoAmI Subscription case. but in case of successful response with no subscription, we call the new parsing method by default, is that what we are looking for?